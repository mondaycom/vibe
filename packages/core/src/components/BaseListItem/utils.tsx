import { EndElement, StartElement } from "./BaseListItem.types";
import { TextType } from "../Text";
import React from "react";
import Avatar from "../Avatar/Avatar";
import styles from "./BaseListItem.module.scss";
import Icon from "../Icon/Icon";
import Text from "../Text/Text";

export function renderSideElement(
  element: StartElement | EndElement,
  disabled: boolean,
  textVariant: TextType
): React.ReactNode {
  switch (element.type) {
    case "avatar":
      return (
        <Avatar
          className={styles.avatar}
          src={element.value}
          withoutBorder
          square={element.square}
          type="img"
          customSize={20}
          disabled={disabled}
        />
      );

    case "icon":
      return <Icon className={styles.icon} icon={element.value} iconSize={20} />;

    case "indent":
      return <div className={styles.indent} />;

    case "suffix":
      return (
        <Text type={textVariant} color="secondary" style={{ maxWidth: "112px" }}>
          {element.value}
        </Text>
      );

    case "custom":
      return element.render();
  }
}
