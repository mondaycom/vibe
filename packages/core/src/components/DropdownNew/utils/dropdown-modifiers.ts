import { Modifier } from "react-popper";

export const matchWidthModifier: Modifier<any>[] = [
  {
    name: "matchWidth",
    enabled: true,
    phase: "beforeWrite",
    requires: ["computeStyles"],
    fn: ({ state }) => {
      if (state.rects && state.rects.reference) {
        state.styles.popper.width = `${state.rects.reference.width}px`;
      }
    }
  }
];
