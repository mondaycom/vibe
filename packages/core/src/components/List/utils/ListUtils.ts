import { MutableRefObject, useState } from "react";
import { ListItemElement } from "../../ListItem/ListItem.types";
import { ListElement } from "../List.types";
import useIsomorphicLayoutEffect from "../../../hooks/ssr/useIsomorphicLayoutEffect";

let listIdCounter = 0;
export const generateListId = () => {
  return `list-${listIdCounter++}`;
};

const VALID_ROLES = ["option", "listitem", "menuitem", "tab", "treeitem"];

export const useListId = (id: string) => {
  const [listId, setListId] = useState<string>();
  useIsomorphicLayoutEffect(() => {
    setListId(id || generateListId());
  }, [id]);
  return listId;
};

export const getListItemIdByIndex = (childrenRefs: MutableRefObject<HTMLElement[]>, index: number): string => {
  return childrenRefs.current[index]?.id;
};

export const getListItemIndexById = (childrenRefs: MutableRefObject<HTMLElement[]>, id: string): number => {
  return childrenRefs.current.findIndex(child => child?.id === id);
};

export const getListItemComponentType = (listComponent: ListElement): ListItemElement => {
  switch (listComponent) {
    case "ul":
    case "ol":
      return "li";
    case "nav":
      return "a";
    default:
      return "div";
  }
};

export const isListItem = (element: HTMLElement) => {
  return element && element instanceof HTMLElement && VALID_ROLES.includes(element.getAttribute("role"));
};

export const getNextListItemIndex = (currentIndex: number, childrenRefs: MutableRefObject<HTMLElement[]>) => {
  while (currentIndex < childrenRefs.current.length - 1) {
    const child = childrenRefs.current[++currentIndex];
    if (isListItem(child)) {
      return currentIndex;
    }
  }
  return undefined;
};

export const getPrevListItemIndex = (currentIndex: number, childrenRefs: MutableRefObject<HTMLElement[]>) => {
  while (currentIndex > 0) {
    const child = childrenRefs.current[--currentIndex];
    if (isListItem(child)) {
      return currentIndex;
    }
  }
  return undefined;
};
