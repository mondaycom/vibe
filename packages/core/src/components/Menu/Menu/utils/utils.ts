import { type ReactElement } from "react";
import { type MenuChild } from "../MenuConstants";

export function isMenuChildSelectable(child: MenuChild): boolean {
  return !!child.type.isSelectable && !child.props.disabled;
}

export const generateMenuItemId = (menuId: string, child: ReactElement, index: number) => {
  return (child.props as { id: string }).id || `${menuId}-item-${index}`;
};
