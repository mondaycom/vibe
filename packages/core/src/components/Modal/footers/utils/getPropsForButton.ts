import { ModalFooterActionProps } from "../ModalFooterBase/ModalFooterBase.types";
import cx from "classnames";

export function getPropsForButton(button?: ModalFooterActionProps, buttonClassName?: string) {
  if (!button) return undefined;
  const { tooltipProps, className, ...rest } = button;
  return {
    ...rest,
    className: tooltipProps?.content ? className : cx(className, buttonClassName),
    tooltipProps: tooltipProps?.content
      ? { ...tooltipProps, referenceWrapperClassName: cx(tooltipProps.referenceWrapperClassName, buttonClassName) }
      : undefined
  };
}
