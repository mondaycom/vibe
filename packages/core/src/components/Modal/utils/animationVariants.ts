import { Variants } from "framer-motion";
import { RefObject } from "react";

const enterEase = [0.0, 0.0, 0.4, 1.0];
const exitEase = [0.6, 0.0, 1.0, 1.0];

export const modalAnimationOverlayVariants: Variants = {
  initial: {
    opacity: 0
  },
  enter: {
    opacity: 1,
    transition: {
      duration: 0.1,
      ease: enterEase
    }
  },
  exit: {
    opacity: 0,
    transition: {
      duration: 0.1,
      ease: exitEase
    }
  }
};

export const modalAnimationCenterPopVariants: Variants = {
  initial: {
    opacity: 1,
    scale: 1,
    x: "-50%",
    y: "-50%"
  },
  enter: {
    opacity: [0, 1, 1],
    scale: [0.8, 1, 1],
    x: "-50%",
    y: "-50%",
    transition: {
      duration: 0.15,
      ease: enterEase,
      times: [0, 0.5, 1]
    }
  },
  exit: {
    opacity: [1, 1, 0],
    scale: [1, 1, 0.8],
    x: "-50%",
    y: "-50%",
    transition: {
      duration: 0.1,
      ease: exitEase,
      times: [0, 0.5, 1]
    }
  }
};

export const modalAnimationAnchorPopVariants: Variants = {
  initial: (anchorElementRef: RefObject<HTMLElement>) => {
    const anchorRect = anchorElementRef.current.getBoundingClientRect();
    const anchorCenterX = anchorRect.left + anchorRect.width / 2;
    const anchorCenterY = anchorRect.top + anchorRect.height / 2;

    const x = `calc(-50% + ${anchorCenterX}px)`;
    const y = `calc(-50% + ${anchorCenterY}px)`;

    return {
      opacity: 0,
      scale: 0.8,
      top: "50%",
      left: "50%",
      x,
      y
    };
  },
  enter: {
    opacity: [0, 0, 1],
    scale: [0.8, 0.8, 1],
    top: "50%",
    left: "50%",
    x: "-50%",
    y: "-50%",
    transition: {
      duration: 0.2,
      ease: enterEase,
      times: [0, 0.4, 1]
    }
  },
  exit: (anchorElementRef: RefObject<HTMLElement>) => {
    const anchorRect = anchorElementRef.current.getBoundingClientRect();
    const anchorCenterX = anchorRect.left + anchorRect.width / 2;
    const anchorCenterY = anchorRect.top + anchorRect.height / 2;

    const x = `calc(-50% + ${anchorCenterX}px)`;
    const y = `calc(-50% + ${anchorCenterY}px)`;

    return {
      opacity: [1, 0, 0],
      scale: [1, 0.8, 0.8],
      top: 0,
      left: 0,
      x,
      y,
      transition: {
        duration: 0.15,
        ease: exitEase,
        times: [0, 0.6, 1]
      }
    };
  }
};

export const modalAnimationFullViewVariants: Variants = {
  enter: {
    opacity: [0.3, 1, 1],
    y: [30, 0],
    transition: {
      duration: 0.25,
      ease: enterEase,
      times: [0, 0.33, 1]
    }
  },
  exit: {
    opacity: 0,
    y: 30,
    transition: {
      duration: 0.1,
      ease: exitEase
    }
  }
};
