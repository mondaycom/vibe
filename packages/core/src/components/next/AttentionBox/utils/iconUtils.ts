import type { SubIcon } from "../../../../types";
import type { AttentionBoxType } from "../AttentionBox.types";
import { ATTENTION_BOX_DEFAULT_ICONS } from "../consts/icons";

export const resolveAttentionBoxIcon = (icon?: SubIcon | false, type?: AttentionBoxType): SubIcon | null => {
  if (icon === false) return null;
  return icon || ATTENTION_BOX_DEFAULT_ICONS[type || "primary"];
};
