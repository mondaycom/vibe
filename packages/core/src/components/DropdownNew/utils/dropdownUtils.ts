import { BaseListItemData } from "../../BaseListItem";
import { ListGroup } from "../../BaseList";

export function normalizeOptions<Item extends BaseListItemData>(
  options: ListGroup<Item>[] | BaseListItemData<Item>[],
  filter?: string,
  filterOption?: (option: Item, inputValue: string) => boolean
): ListGroup<Item>[] {
  let indexCounter = 0;
  const defaultFilterFn = (item: Item, inputValue: string) =>
    !inputValue || item.label.toLowerCase().includes(inputValue.toLowerCase());
  const currentFilterFn = filterOption || defaultFilterFn;

  return Array.isArray(options) && options.some(item => "options" in item)
    ? (options as ListGroup<Item>[]).flatMap(group => {
        const filteredOptions = group.options
          .filter(item => currentFilterFn(item, filter || ""))
          .map(item => ({ ...item, index: indexCounter++ }));

        return filteredOptions.length > 0 ? [{ ...group, options: filteredOptions }] : [];
      })
    : [
        {
          label: undefined,
          options: (options as BaseListItemData<Item>[])
            .filter(item => currentFilterFn(item, filter || ""))
            .map(item => ({ ...item, index: indexCounter++ }))
        }
      ];
}
