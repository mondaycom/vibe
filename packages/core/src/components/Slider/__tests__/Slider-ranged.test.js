import React from "react";
import { act } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { snapshotDiff } from "../../../utils/jest-utils";
import { renderSliderInRangeMode } from "./sliderTestUtils";

vi.useFakeTimers();

vi.mock("../../TextField/TextField", () => {
  return props => {
    return <div data-testid="mock-text-field-comp">{JSON.stringify(props)}</div>;
  };
});

it("a. Ranges Slider: basic renderer", async () => {
  const { asFragment } = await renderSliderInRangeMode();
  expect(asFragment().firstChild).toMatchSnapshot();
});

describe("b. Ranges Slider Active/Inactive", () => {
  it("01. show Tooltip when hover Start Thumb", async () => {
    let before;
    let after;
    await act(async () => {
      const { asFragment, elThumbStart } = await renderSliderInRangeMode();
      before = asFragment().firstChild;
      userEvent.hover(elThumbStart);
      vi.advanceTimersByTime(999);
      after = asFragment().firstChild;
    });
    expect(snapshotDiff(before, after)).toMatchSnapshot();
  });

  it("02. show Tooltip when hover End Thumb", async () => {
    let before;
    let after;
    await act(async () => {
      const { asFragment, elThumbEnd } = await renderSliderInRangeMode();
      before = asFragment().firstChild;
      userEvent.hover(elThumbEnd);
      vi.advanceTimersByTime(999);
      after = asFragment().firstChild;
    });
    expect(snapshotDiff(before, after)).toMatchSnapshot();
  });

  it("03. hide Tooltip when unhover Thumb", async () => {
    let before;
    let after;
    await act(async () => {
      const { asFragment, elThumbStart } = await renderSliderInRangeMode();
      userEvent.hover(elThumbStart);
      vi.advanceTimersByTime(999);
      before = asFragment().firstChild;
      userEvent.unhover(elThumbStart);
      vi.advanceTimersByTime(999);
      after = asFragment().firstChild;
    });
    expect(snapshotDiff(before, after)).toMatchSnapshot();
  });

  it("04. activate Start Thumb by Tab key press ", async () => {
    let before;
    let after;
    await act(async () => {
      const { asFragment } = await renderSliderInRangeMode();
      before = asFragment().firstChild;
      userEvent.tab();
      vi.advanceTimersByTime(999);
      after = asFragment().firstChild;
    });
    expect(snapshotDiff(before, after)).toMatchSnapshot();
  });

  it("05. activate End Thumb by Tab key press ", async () => {
    let before;
    let after;
    await act(async () => {
      const { asFragment } = await renderSliderInRangeMode();
      userEvent.tab();
      vi.advanceTimersByTime(999);
      before = asFragment().firstChild;
      userEvent.tab();
      vi.advanceTimersByTime(999);
      after = asFragment().firstChild;
    });
    expect(snapshotDiff(before, after)).toMatchSnapshot();
  });

  it("06. de-activate Thumb by Tab key press, when End Thumb focused ", async () => {
    let before;
    let after;
    await act(async () => {
      const { asFragment, elThumbEnd } = await renderSliderInRangeMode();
      elThumbEnd.focus();
      before = asFragment().firstChild;
      userEvent.tab();
      vi.advanceTimersByTime(999);
      after = asFragment().firstChild;
    });
    expect(snapshotDiff(before, after)).toMatchSnapshot();
  });
});

describe("c. Ranges Slider Key Events", () => {
  it("01. decrease value by Left/Down keys press", async () => {
    let before;
    let after;
    await act(async () => {
      const { asFragment, elThumbStart } = await renderSliderInRangeMode({ showValue: true });
      elThumbStart.focus();
      before = asFragment().firstChild;
      userEvent.keyboard("{arrowleft}");
      userEvent.keyboard("{arrowdown}");
      after = asFragment().firstChild;
    });
    expect(snapshotDiff(before, after)).toMatchSnapshot();
  });

  it("02. increase value by Right/Up keys press", async () => {
    let before;
    let after;
    await act(async () => {
      const { asFragment, elThumbEnd } = await renderSliderInRangeMode({ showValue: true });
      elThumbEnd.focus();
      before = asFragment().firstChild;
      userEvent.keyboard("{arrowright}");
      userEvent.keyboard("{arrowup}");
      after = asFragment().firstChild;
    });
    expect(snapshotDiff(before, after)).toMatchSnapshot();
  });

  it("03. decrease value (step 10%) by PageDown key press", async () => {
    let before;
    let after;
    await act(async () => {
      const { asFragment, elThumbEnd } = await renderSliderInRangeMode({ showValue: true });
      elThumbEnd.focus();
      before = asFragment().firstChild;
      userEvent.keyboard("{pagedown}");
      after = asFragment().firstChild;
    });
    expect(snapshotDiff(before, after)).toMatchSnapshot();
  });

  it("04. increase value (step 10%) by PageUp key pres", async () => {
    let before;
    let after;
    await act(async () => {
      const { asFragment, elThumbStart } = await renderSliderInRangeMode({ showValue: true });
      elThumbStart.focus();
      before = asFragment().firstChild;
      userEvent.keyboard("{pageup}");
      userEvent.keyboard("{pageup}");
      after = asFragment().firstChild;
    });
    expect(snapshotDiff(before, after)).toMatchSnapshot();
  });

  it("05. change value by keys press in Step and Label mode", async () => {
    let before;
    let after;
    await act(async () => {
      const { asFragment, elThumbStart } = await renderSliderInRangeMode({
        indicateSelection: true,
        showValue: true,
        step: 10
      });
      elThumbStart.focus();
      before = asFragment().firstChild;
      await userEvent.keyboard("{arrowright}");
      await userEvent.keyboard("{arrowright}");
      after = asFragment().firstChild;
    });
    expect(snapshotDiff(before, after)).toMatchSnapshot();
  });
});
