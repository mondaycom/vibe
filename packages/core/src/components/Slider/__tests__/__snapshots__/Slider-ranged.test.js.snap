// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a. Ranges Slider: basic renderer 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 25%; width: 40%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="25"
        aria-valuetext="25%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 25%;"
        tabindex="0"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="65"
        aria-valuetext="65%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-1"
        role="slider"
        style="left: 65%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`b. Ranges Slider Active/Inactive 01. show Tooltip when hover Start Thumb 1`] = `
Snapshot Diff:
- Default Render
+ Current Render

@@ --- --- @@
          style="left: 25%;"
          tabindex="0"
        />
+       <span
+         class="monday-style-dialog-content-wrapper contentWrapper"
+         data-popper-reference-hidden="false"
+         data-testid="tooltip"
+         style="position: absolute; left: 0px; top: 0px;"
+       >
+         <div
+           class="contentComponent top hasTooltip"
+         >
+           <div
+             class="tooltip dark"
+           >
+             <div
+               class="content"
+             >
+               25%
+             </div>
+           </div>
+           <div
+             class="arrow arrow dark"
+             data-placement="top"
+             style="position: absolute;"
+           />
+         </div>
+       </span>
        <div
          aria-disabled="false"
          aria-valuemax="100"
`;

exports[`b. Ranges Slider Active/Inactive 02. show Tooltip when hover End Thumb 1`] = `
Snapshot Diff:
- Default Render
+ Current Render

@@ --- --- @@
          style="left: 65%;"
          tabindex="0"
        />
+       <span
+         class="monday-style-dialog-content-wrapper contentWrapper"
+         data-popper-reference-hidden="false"
+         data-testid="tooltip"
+         style="position: absolute; left: 0px; top: 0px;"
+       >
+         <div
+           class="contentComponent top hasTooltip"
+         >
+           <div
+             class="tooltip dark"
+           >
+             <div
+               class="content"
+             >
+               65%
+             </div>
+           </div>
+           <div
+             class="arrow arrow dark"
+             data-placement="top"
+             style="position: absolute;"
+           />
+         </div>
+       </span>
      </div>
    </div>
  </div>
`;

exports[`b. Ranges Slider Active/Inactive 03. hide Tooltip when unhover Thumb 1`] = `
Snapshot Diff:
- Default Render
+ Current Render

@@ --- --- @@
          style="left: 25%;"
          tabindex="0"
        />
-       <span
-         class="monday-style-dialog-content-wrapper contentWrapper"
-         data-popper-reference-hidden="false"
-         data-testid="tooltip"
-         style="position: absolute; left: 0px; top: 0px;"
-       >
-         <div
-           class="contentComponent top hasTooltip"
-         >
-           <div
-             class="tooltip dark"
-           >
-             <div
-               class="content"
-             >
-               25%
-             </div>
-           </div>
-           <div
-             class="arrow arrow dark"
-             data-placement="top"
-             style="position: absolute;"
-           />
-         </div>
-       </span>
        <div
          aria-disabled="false"
          aria-valuemax="100"
`;

exports[`b. Ranges Slider Active/Inactive 04. activate Start Thumb by Tab key press  1`] = `
Snapshot Diff:
- Default Render
+ Current Render

@@ --- --- @@
          style="left: 25%;"
          tabindex="0"
        />
+       <span
+         class="monday-style-dialog-content-wrapper contentWrapper"
+         data-popper-reference-hidden="false"
+         data-testid="tooltip"
+         style="position: absolute; left: 0px; top: 0px;"
+       >
+         <div
+           class="contentComponent top hasTooltip"
+         >
+           <div
+             class="tooltip dark"
+           >
+             <div
+               class="content"
+             >
+               25%
+             </div>
+           </div>
+           <div
+             class="arrow arrow dark"
+             data-placement="top"
+             style="position: absolute;"
+           />
+         </div>
+       </span>
        <div
          aria-disabled="false"
          aria-valuemax="100"
`;

exports[`b. Ranges Slider Active/Inactive 05. activate End Thumb by Tab key press  1`] = `
Snapshot Diff:
- Default Render
+ Current Render

@@ --- --- @@
          aria-valuemin="0"
          aria-valuenow="25"
          aria-valuetext="25%"
-         class="thumb small notDisabledThumb"
+         class="thumb small focused notDisabledThumb"
          data-testid="monday-slider__thumb-0"
          role="slider"
          style="left: 25%;"
+         tabindex="0"
+       />
+       <div
+         aria-disabled="false"
+         aria-valuemax="100"
+         aria-valuemin="0"
+         aria-valuenow="65"
+         aria-valuetext="65%"
+         class="thumb small focused notDisabledThumb"
+         data-testid="monday-slider__thumb-1"
+         role="slider"
+         style="left: 65%;"
          tabindex="0"
        />
        <span
@@ --- --- @@
              <div
                class="content"
              >
-               25%
+               65%
              </div>
            </div>
            <div
@@ --- --- @@
            />
          </div>
        </span>
-       <div
-         aria-disabled="false"
-         aria-valuemax="100"
-         aria-valuemin="0"
-         aria-valuenow="65"
-         aria-valuetext="65%"
-         class="thumb small notDisabledThumb"
-         data-testid="monday-slider__thumb-1"
-         role="slider"
-         style="left: 65%;"
-         tabindex="0"
-       />
      </div>
    </div>
  </div>
`;

exports[`b. Ranges Slider Active/Inactive 06. de-activate Thumb by Tab key press, when End Thumb focused  1`] = `
Snapshot Diff:
- Default Render
+ Current Render

@@ --- --- @@
          aria-valuemin="0"
          aria-valuenow="25"
          aria-valuetext="25%"
-         class="thumb small focused notDisabledThumb"
+         class="thumb small notDisabledThumb"
          data-testid="monday-slider__thumb-0"
          role="slider"
          style="left: 25%;"
@@ --- --- @@
          aria-valuemin="0"
          aria-valuenow="65"
          aria-valuetext="65%"
-         class="thumb small focused notDisabledThumb"
+         class="thumb small notDisabledThumb"
          data-testid="monday-slider__thumb-1"
          role="slider"
          style="left: 65%;"
          tabindex="0"
        />
-       <span
-         class="monday-style-dialog-content-wrapper contentWrapper"
-         data-popper-reference-hidden="false"
-         data-testid="tooltip"
-         style="position: absolute; left: 0px; top: 0px;"
-       >
-         <div
-           class="contentComponent top hasTooltip expandAppear expandAppearActive"
-         >
-           <div
-             class="tooltip dark"
-           >
-             <div
-               class="content"
-             >
-               65%
-             </div>
-           </div>
-           <div
-             class="arrow arrow dark"
-             data-placement="top"
-             style="position: absolute;"
-           />
-         </div>
-       </span>
      </div>
    </div>
  </div>
`;

exports[`c. Ranges Slider Key Events 01. decrease value by Left/Down keys press 1`] = `
Snapshot Diff:
- Default Render
+ Current Render

@@ --- --- @@
        />
        <div
          class="filledTrack"
-         style="left: 25%; width: 40%;"
+         style="left: 23%; width: 42%;"
        />
        <div
          aria-disabled="false"
          aria-valuemax="100"
          aria-valuemin="0"
-         aria-valuenow="25"
-         aria-valuetext="25%"
+         aria-valuenow="23"
+         aria-valuetext="23%"
          class="thumb small notDisabledThumb"
          data-testid="monday-slider__thumb-0"
          role="slider"
-         style="left: 25%;"
+         style="left: 23%;"
          tabindex="0"
        >
          <label
            class="label colorPrimary positionTop"
          >
-           25%
+           23%
          </label>
        </div>
        <div
`;

exports[`c. Ranges Slider Key Events 02. increase value by Right/Up keys press 1`] = `
Snapshot Diff:
- Default Render
+ Current Render

@@ --- --- @@
        />
        <div
          class="filledTrack"
-         style="left: 25%; width: 40%;"
+         style="left: 25%; width: 42%;"
        />
        <div
          aria-disabled="false"
@@ --- --- @@
          aria-disabled="false"
          aria-valuemax="100"
          aria-valuemin="0"
-         aria-valuenow="65"
-         aria-valuetext="65%"
+         aria-valuenow="67"
+         aria-valuetext="67%"
          class="thumb small focused notDisabledThumb"
          data-testid="monday-slider__thumb-1"
          role="slider"
-         style="left: 65%;"
+         style="left: 67%;"
          tabindex="0"
        >
          <label
            class="label colorPrimary positionTop"
          >
-           65%
+           67%
          </label>
        </div>
      </div>
`;

exports[`c. Ranges Slider Key Events 03. decrease value (step 10%) by PageDown key press 1`] = `
Snapshot Diff:
- Default Render
+ Current Render

@@ --- --- @@
        />
        <div
          class="filledTrack"
-         style="left: 25%; width: 40%;"
+         style="left: 25%; width: 30%;"
        />
        <div
          aria-disabled="false"
@@ --- --- @@
          aria-disabled="false"
          aria-valuemax="100"
          aria-valuemin="0"
-         aria-valuenow="65"
-         aria-valuetext="65%"
+         aria-valuenow="55"
+         aria-valuetext="55%"
          class="thumb small focused notDisabledThumb"
          data-testid="monday-slider__thumb-1"
          role="slider"
-         style="left: 65%;"
+         style="left: 55%;"
          tabindex="0"
        >
          <label
            class="label colorPrimary positionTop"
          >
-           65%
+           55%
          </label>
        </div>
      </div>
`;

exports[`c. Ranges Slider Key Events 04. increase value (step 10%) by PageUp key pres 1`] = `
Snapshot Diff:
- Default Render
+ Current Render

@@ --- --- @@
        />
        <div
          class="filledTrack"
-         style="left: 25%; width: 40%;"
+         style="left: 45%; width: 20%;"
        />
        <div
          aria-disabled="false"
          aria-valuemax="100"
          aria-valuemin="0"
-         aria-valuenow="25"
-         aria-valuetext="25%"
+         aria-valuenow="45"
+         aria-valuetext="45%"
          class="thumb small notDisabledThumb"
          data-testid="monday-slider__thumb-0"
          role="slider"
-         style="left: 25%;"
+         style="left: 45%;"
          tabindex="0"
        >
          <label
            class="label colorPrimary positionTop"
          >
-           25%
+           45%
          </label>
        </div>
        <div
`;

exports[`c. Ranges Slider Key Events 05. change value by keys press in Step and Label mode 1`] = `
Snapshot Diff:
- Default Render
+ Current Render

@@ --- --- @@
      <div
        data-testid="mock-text-field-comp"
      >
-       {"value":"25%","className":"selectionIndicator"}
+       {"value":"45%","className":"selectionIndicator"}
      </div>
    </div>
    <div
@@ --- --- @@
        />
        <div
          class="filledTrack"
-         style="left: 25%; width: 40%;"
+         style="left: 45%; width: 20%;"
        />
        <div
          aria-disabled="false"
          aria-valuemax="100"
          aria-valuemin="0"
-         aria-valuenow="25"
-         aria-valuetext="25%"
+         aria-valuenow="45"
+         aria-valuetext="45%"
          class="thumb small notDisabledThumb"
          data-testid="monday-slider__thumb-0"
          role="slider"
-         style="left: 25%;"
+         style="left: 45%;"
          tabindex="0"
        >
          <label
            class="label colorPrimary positionTop"
          >
-           25%
+           45%
          </label>
        </div>
        <div
`;
